Tests/SwiftParserTest/ThenStatementTests.swift

```diff
@@ -685,7 +673,10 @@ final class ThenStatementTests: ParserTestCase {
       then
         .foo
       """,
-      substructure: ThenStmtSyntax(expression: MemberAccessExprSyntax(name: .identifier("foo")))
+      substructure: MemberAccessExprSyntax(
+        base: DeclReferenceExprSyntax(baseName: .identifier("then")),
+        name: .identifier("foo")
```

Comment:
Hmm, this does conflict [with the proposal](https://github.com/swiftlang/swift-evolution/blob/7bb5b9d6e02caaf2e7dbd9e9c084ce969be66825/proposals/NNNN-then-keyword.md#parsing-ambiguities-with-then):

> `then.foo` is a member access, `then .foo` is a then statement, as is:
> 
> ```swift
> then
>   .member
> ```
> 
> If member access was still desired, back ticks could be used:
> 
> ```swift
> `then`
>   .member
> ```
> 
> This is a potential (albeit unlikely) source break, but the back tick fix can be applied to the 5.9 compiler today to ensure existing code can compile with both the old and new compiler.

IIRC it was felt that `then` would much more likely be a function name than a variable name, which made the member interpretation fairly unlikely